sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -268,48 +334,52 @@ static int get_operation_status()
          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
         != MQTTCLIENT_SUCCESS)
     {
-      printf("Failed to receive message, return code %d\n", rc);
+      printf("Failed to receive message, MQTTClient return code %d\n", rc);
       return rc;
     }
 
+    printf("Received a message from service.\n");
+    printf("Topic: %s\n", topic);
+
     if (topic_len == 0)
     {
       // The length of the topic if there are one or more NULL characters embedded in topic,
       // otherwise topic_len is 0.
       topic_len = (int)strlen(topic);
     }
 
-    printf("Message arrived\n");
-    printf("     topic (%d): %s\n", topic_len, topic);
-    printf("   message (%d): ", message->payloadlen);
-    print_az_span(az_span_init(message->payload, message->payloadlen));
-
-    az_iot_provisioning_client_register_response response;
+    az_span topic_span = az_span_init((uint8_t*)topic, topic_len);
+    az_span message_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
 
     // Parse the incoming message and payload
+    az_iot_provisioning_client_register_response response;
     if (az_failed(
             rc = az_iot_provisioning_client_parse_received_topic_and_payload(
-                &provisioning_client,
-                az_span_init((uint8_t*)topic, topic_len),
-                az_span_init((uint8_t*)message->payload, message->payloadlen),
-                &response)))
+                &provisioning_client, topic_span, message_span, &response)))
     {
-      printf("Message from unknown topic, return code %d\n", rc);
+      printf("Message from unknown topic, az_result return code %0x4\n", rc);
+      MQTTClient_freeMessage(&message);
```

Comment:
Instead of having the free everywhere, wrap the processing function in a separate function. That should allow less duplicated code.

Reply:
I had considered putting it all together, something like "return rc_free(rc, &message, topic)" so that there is just one line.  But not sure if that would be harder on the sample reader to jump around.

Reply:
will refactor method in future PR.